crates/solidity/outputs/cargo/tests/src/doc_examples/using_queries.rs

```diff
@@ -0,0 +1,176 @@
+use std::path::Path;
+
+use anyhow::Result;
+use infra_utils::paths::PathExtensions;
+use semver::Version;
+use slang_solidity::kinds::RuleKind;
+use slang_solidity::language::Language;
+use slang_solidity::parse_output::ParseOutput;
+use slang_solidity::query::Query;
+
+fn parse_doc_input_file<T: AsRef<Path>>(path: T) -> Result<ParseOutput> {
+    let input_path =
+        Path::repo_path(Path::new("documentation/public/user-guide/inputs").join(path.as_ref()));
+
+    let input_path = input_path.unwrap_str();
+
+    let source = std::fs::read_to_string(input_path)?;
```

Comment:
nit: also existing helper:

```suggestion
    let source = input_path.read_to_string().trim();
```

Reply:
Not a fan of using our custom method helper when `std::fs` has exactly that already but done

Reply:
I found it a helpful pattern in chaining calls (`Path(...).join(...).read_to_string().lines().etc(....)`), and I ended up using it everywhere for consistency.
But completely up to you if you prefer otherwise. Just pointing it out in case you find it helpful as well.

Reply:
> when std::fs has exactly that 

@Xanewok  I remembered the reason I added `read_to_string()` (and other `Path` extensions) in the first place. All of these `std::fs` functions don't print helpful error messages when something fails. The error doesn't even include the file path it was trying to read/write/etc..., and I found myself having to repeat `.with_context(...more info...)` tens of times, so I extracted these context errors to a shared place. I should add some comments to `PathExtensions` next.